odl/util/numerics.py

```diff
@@ -0,0 +1,152 @@
+﻿# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Numerical helper functions for convenience or speed."""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+
+from future import standard_library
+standard_library.install_aliases()
+
+# External module imports
+
+
+__all__ = ('apply_on_boundary',)
+
+
+def apply_on_boundary(array, func, only_once=True, which_boundaries=None,
+                      axis_order=None):
+    """Apply a function of the boundary of an n-dimensional array.
```

Comment:
I was more thinking of this as a general utility and adhering to some "standard", by providing an optional out parameter and returning the result this could be generalized and have the same "style" as we aim for.